public/js/reducers/sources.js

```diff
@@ -81,13 +79,19 @@ function update(state = initialState, action) {
   return state;
 }
 
-function getFilenameFromUrl(sourceUrl) {
+function _updateSource(source) {
+  const pathname = getPathnameFromUrl(source.url);
+  const filename = pathname.substring(pathname.lastIndexOf("/") + 1);
+  return Object.assign({}, source, { pathname, filename });
+}
+
+function getPathnameFromUrl(sourceUrl) {
   if (!sourceUrl) {
     return "";
   }
 
   const url = new URL(sourceUrl);
```

Comment:
I'm convinced now that that should live somewhere else and we should make a selector that extracts the filename from a source. The `URL` is undefined in node and none of this code should depend on the browser.